packages/ui/src/components/va-chip/VaChip.vue

```diff
@@ -35,139 +35,136 @@
-  color = prop<string>({ type: String, default: 'primary' })
-  outline = prop<boolean>({ type: Boolean, default: false })
-  flat = prop<boolean>({ type: Boolean, default: false })
-  icon = prop<string>({ type: String, default: '' })
-  disabled = prop<boolean>({ type: Boolean, default: false })
-  square = prop<boolean>({ type: Boolean, default: false })
-  tag = prop<string>({ type: String, default: 'span' })
-  size = prop<string>({
-    type: String,
-    default: 'medium',
-    validator: (value: string) => {
-      return ['medium', 'small', 'large'].includes(value)
-    },
-  })
-  shadow = prop<boolean>({ type: Boolean, default: false })
-}
-
-const ChipPropsMixin = Vue.with(ChipProps)
-
-@Options({
+export default defineComponent({
   name: 'VaChip',
+
   components: { VaIcon },
-  emits: ['focus'],
-})
-export default class VaChip extends mixins(
-  RouterLinkMixin,
-  StatefulMixin,
-  ColorMixin,
-  ChipPropsMixin,
-) {
-  SetupContext = setup(() => {
+
+  emits: [...useStatefulEmits, 'focus'],
+
+  props: {
+    ...useRouterLinkProps,
+    ...useColorProps,
+    ...useStatefulProps,
+    modelValue: { type: Boolean as PropType<boolean>, default: true },
+    closeable: { type: Boolean as PropType<boolean>, default: false },
+    outline: { type: Boolean as PropType<boolean>, default: false },
+    disabled: { type: Boolean as PropType<boolean>, default: false },
+    square: { type: Boolean as PropType<boolean>, default: false },
+    shadow: { type: Boolean as PropType<boolean>, default: false },
+    flat: { type: Boolean as PropType<boolean>, default: false },
+    icon: { type: String as PropType<string>, default: '' },
+    tag: { type: String as PropType<string>, default: 'span' },
+
+    size: {
+      type: String as PropType<string>,
+      default: 'medium',
+      validator: (value: string) => {
+        return ['small', 'medium', 'large'].includes(value)
+      },
+    },
+  },
+
+  setup (props, { emit }) {
     const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
 
-    return {
-      hasKeyboardFocus,
-      keyboardFocusListeners,
-    }
-  })
+    const { getColor } = useColors()
+    const colorComputed = computed(() => getColor(props.color))
+    const borderColor = computed(() => props.outline ? colorComputed : '')
 
-  hoverState = false
-  focusState = false
+    const { tagComputed, hrefComputed } = useRouterLink(props)
 
-  created () {
-    watch(() => this.hoverState, (value) => {
-      this.updateHoverState(value)
-    })
-  }
+    const hoverState = ref(false)
+    const updateHoverState = (isHover: boolean) => {
+      hoverState.value = isHover
+    }
 
-  get iconSize () {
-    const size: any = {
+    const size = {
       small: '0.875rem',
       medium: '1rem',
       large: '1.25rem',
-    }
-    return size[this.size]
-  }
+    } as Record<string, string>
+    const iconSize = computed(() => size[props.size])
```

Comment:
Sad to see size value doesn't connected with SizeConfig. Let's have some TODO here on new issue.

Reply:
https://github.com/epicmaxco/vuestic-ui/issues/1671